packages/validation/src/utils/getRule.ts

```diff
@@ -18,10 +19,10 @@ export function getRuleForResult(response: ValidationResponse, result: Validatio
 }
 
 export function getRuleForResultV2(run: ValidationRun | undefined, result: ValidationResult): RuleMetadata {
-  const toolPluginIndex = result.rule.toolComponent.index;
+  const toolPluginIndex = isDefined(result.rule.toolComponent.index) ? result.rule.toolComponent.index : -1;
```

Comment:
This works just fine, but you could also use `= result.rule.toolComponent.index ?? -1` (just to be consistent with the rest of it)